superset-frontend/src/views/datasetList/DatasetModal.tsx

```diff
@@ -0,0 +1,318 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import styled from '@superset-ui/style';
+import { SupersetClient } from '@superset-ui/connection';
+import { t } from '@superset-ui/translation';
+import Icon from 'src/components/Icon';
+import Select from 'src/components/Select';
+import Modal from './Modal';
+
+import withToasts from '../../messageToasts/enhancers/withToasts';
+
+type option = {
+  label: string;
+  value: string;
+};
+
+interface Props {
+  addDangerToast: (msg: string) => void;
+  addSuccessToast: (msg: string) => void;
+  onChange: any;
+  onDatasourceSave: any;
+  onHide: any;
```

Comment:
can we do better than `any` here? It looks like they're functions, so we should at least be able to say `() => unknown` i'd think. or maybe define actual arguments too.

A final note, in general it's better to use `unknown` vs. `any` when you're unsure of a type, because `any` completely removes any type safety from code that uses the prop.

Reply:
++ to what @etr2460 said. TypeScript will actually start complaining about explicit anys at some point. They already emit warnings in `superset-ui`.

Reply:
@etr2460 @ktmud actually `onChange` and  `onDatasourceSave` are not called anywhere and I will remove them. Sorry for the confusion.